### ISSUE TITLE
deployApps.sh requires domain_home to be passed even in online mode

### ISSUE THREAD

**Issue by cchaplin** [oracle/weblogic-deploy-tooling#257]
-domain_home parameter is mandatory for deployApps.sh even when using online mode. Per the documentation, 

> In WLST online mode, simply replace the -domain_home argument with the information on how to connect to the WebLogic Server Administration Server

But when I try the command without -domain_home, the script throws error

> ####<Feb 25, 2019 4:41:40 PM> <SEVERE> <deploy> <main> <WLSDPLY-20008> <deployApps argument processing failed: deployApps invoked with missing required argument: -domain_home>

Command used:
`weblogic-deploy/bin/deployApps.sh -oracle_home /home/wl005/Oracle/Middleware -domain_type WLS  -archive_file my_app.zip -model_file app_dep.yml -admin_url t3://localhost:7001 -admin_user wlsadmin
`

**Comment by rakillen** [comment]
I'm not clear if we should just fix the documentation to say the argument is required, or allow it to be omitted.

For deployApps and updateDomain, the domain home is used to resolve @@DOMAIN_HOME@@ tokens in the model, and to unzip and configure artifacts from the archive zip. We can require it only if the archive zip is specified, but we can't tell if the token is needed before parsing the model.

Another option is to get the domain home from the RootDirectory property in WLST, if the argument is not specified. This could mislead a user into thinking the archive artifacts will deploy to a remote file system, but that error would be caught and logged.

**Comment by evgeny-pluzhnikov** [comment]
Guys,
Please fix your docs as a starting point. It's really hard to follow your documentation when your examples don't match the content of the doc and when real behaviour does not match the doc either. BTW do you actually need domain home for online deployments?

**Comment by ddsharpe** [comment]
WDT uses the local folders for the OH and domain home.  Online mode is not intended to be used remotely.  Online mode is intended to allow updates without restarting the server.

**Comment by evgeny-pluzhnikov** [comment]
Can you fix the doc then?

**Comment by ddsharpe** [comment]
@evgeny-pluzhnikov Please be more specific.  What is it that you would like fixed in the documentation?

**Comment by evgeny-pluzhnikov** [comment]
Sure.
https://github.com/oracle/weblogic-deploy-tooling/blob/master/site/deploy.md

The following statement is incorrect. Example does not match the statement. You suggest that you can skip setting domain_home. Not true.

Here:

In WLST online mode, simply **replace the -domain_home argument with the information on how to connect to the WebLogic Server Administration Server**, for example:

weblogic-deploy\bin\deployApps.cmd -oracle_home c:\wls12213 **-domain_home domains\DemoDomain** -archive_file DemoDomain.zip -mod
